portal-ui/src/screens/CreatePassword/index.tsx

```diff
@@ -0,0 +1,269 @@
+import {makeStyles} from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+import Copyright from "../../components/Copyright";
+import request from "superagent";
+import {useHistory} from "react-router";
+import {CircularProgress} from "@material-ui/core";
+import storage from 'local-storage-fallback';
+
+const useStyles = makeStyles(theme => ({
+    '@global': {
+        body: {
+            backgroundColor: theme.palette.common.white,
+        },
+    },
+    paper: {
+        marginTop: theme.spacing(8),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    avatar: {
+        margin: theme.spacing(1),
+        backgroundColor: theme.palette.secondary.main,
+    },
+    form: {
+        width: '100%', // Fix IE 11 issue.
+        marginTop: theme.spacing(3),
+    },
+    submit: {
+        margin: theme.spacing(3, 0, 2),
+    },
+    errorBlock: {
+        color: 'red',
+    },
+    spinner: {
+        margin: 'auto',
+
+    }
+}));
+const CreatePassword: React.FC = () => {
+    const classes = useStyles();
+    const {push} = useHistory()
+    const [password, setPassword] = React.useState<string>('');
+    const [token, setToken] = React.useState<string>('');
+    const [tokenValidated, setTokenValidated] = React.useState<boolean>(false);
+    const [tokenValid, setTokenValid] = React.useState<boolean>(true);
+    const [repeatPassword, setRepeatPassword] = React.useState<string>('');
+    const [error, setError] = React.useState<string>('');
+
+    const formSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        if (password != repeatPassword) {
+            setError("Passwords don't match");
+            return
+        }
+        if (password.length < 8) {
+            setError("Password should longer than 8 characters");
+            return
+        }
+        const url = '/api/v1/users/set_password';
+        request
+            .post(url)
+            .send({
+                url_token: token,
+                password: password,
+            })
+            .then((res: any) => {
+                if (res.body.jwt_token) {
+                    // store the jwt token
+                    storage.setItem('token', res.body.jwt_token);
+                    return res.body.jwt_token;
+                } else if (res.body.error) {
+                    // throw will be moved to catch block once bad CreatePassword returns 403
+                    throw res.body.error;
+                }
+            }).then(() => {
+            push('/dashboard');
+        })
+            .catch(err => {
+                setError(err);
+            });
+    }
+
+    // validate the token passed via url
+
+    React.useEffect(() => {
+        const urlParams = new URLSearchParams(window.location.search);
+        const urlToken = urlParams.get("t") as string;
+        if (urlToken == null || urlToken == '') {
```

Comment:
Can we use triple equals here? We are getting a warning here:

```
  Line 115:42:  Expected '===' and instead saw '=='  eqeqeq
```

```suggestion
        if (urlToken === null || urlToken === '') {
```